### DISCUSSION TITLE
Support MacCatalyst Builds

### DISCUSSION THREAD

**Discussion by Sherlouk** [SwiftPackageIndex/SwiftPackageIndex-Server#1363] [Ideas]
I suspect this one might get shot down quickly, which is totally cool with me, but it could be nice to add a new build target for macOS via Catalyst.

My new [Swift Package](https://swiftpackageindex.com/Sherlouk/SpanGrid) supports iOS, but not macOS, and until I did a very simple patch a couple minutes ago didn't support Catalyst either. The SPI build matrix didn't highlight that fact, and perhaps it could have?

**Comment by daveverwer** [comment]
🔫 

😂

**Comment by Sherlouk** [comment]
Water pistol is really not that menacing 😂 Genuinely interested though is it truly off the cards?

**Comment by daveverwer** [comment]
> Water pistol is not that menacing 😂

That was intentional 👍

I can see a good argument for adding Catalyst as a platform and I do think it'd be worth adding as a separate platform rather than trying to aggregate the macOS compatibility into (native macOS OR macCatalyst) because they are so fundamentally different.

However, in terms of priorities, I can't say where it'd lie right now. It wouldn't be at the top of my current list, and given the fact that the implementation would lie inside the build system which isn't super easy to open-source given the close integration with our deployment environment, it would need to be primarily @finestructure who implements it.

**Comment by mattmassicotte** [comment]
Given that Catalyst is a currently shipping Apple technology *and* that it can be quite hard for a developer to determine if a package has Catalyst support, I think this could be a real benefit.

**Comment by finestructure** [comment]
Does anyone know what the `xcodebuild` command for MacCatalyst would look like? For reference, we have:

- `ios`

```
xcodebuild build -scheme "${scheme}" -destination "generic/platform=ios"
```

- `macos-xcodebuild`

```
xcodebuild build -scheme "${scheme}" -destination 'platform=macos,arch=x86_64'
```

**Reply by Sherlouk** [reply]
* `macos-catalyst`
```
xcodebuild build -scheme "${scheme}" -destination 'platform=macos,variant=Mac Catalyst'
```

**Comment by Sherlouk** [comment]
`-destination 'platform=macOS,variant=Mac Catalyst'` if I recall correctly

**Comment by Sherlouk** [comment]
According to https://developer.apple.com/forums/thread/120542 you can set arch too, probably makes sense.

```
xcodebuild build -scheme "${scheme}" -destination 'platform=macos,arch=x86_64,variant=Mac Catalyst'
```

**Comment by mattmassicotte** [comment]
I thought I'd revive this. I still think this makes a lot of sense. Catalyst remains incredibly difficult to support in certain situations and having glanceable information for developers searching for packages would really be amazing.

**Reply by rakuyoMo** [reply]
I also think it would be great if this feature could be supported.

**Reply by rolson** [reply]
This would be a big help for our package as well.
